Apps/Contoso.WPF.Demo/App.xaml.cs

```diff
@@ -16,7 +18,49 @@ public partial class App
         protected override void OnStartup(StartupEventArgs e)
         {
             AppCenter.LogLevel = LogLevel.Verbose;
+
+            // User callbacks.
+            Crashes.ShouldAwaitUserConfirmation = ConfirmationHandler;
+            Crashes.ShouldProcessErrorReport = (report) =>
+            {
+                Log($"Determining whether to process error report with an ID: {report.Id}");
+                return true;
+            };
+
+            // Event handlers.
+            Crashes.SendingErrorReport += (_, args) => Log($"Sending error report for an error ID: {args.Report.Id}");
+            Crashes.SentErrorReport += (_, args) => Log($"Sent error report for an error ID: {args.Report.Id}");
+            Crashes.FailedToSendErrorReport += (_, args) => Log($"Failed to send error report for an error ID: {args.Report.Id}");
+
+            // Start AppCenter.
             AppCenter.Start("f4e2a83d-3052-4884-8176-8b2c50277d16", typeof(Analytics), typeof(Crashes));
+            Crashes.HasCrashedInLastSessionAsync().ContinueWith(hasCrashed =>
+            {
+                Log("Crashes.HasCrashedInLastSession=" + hasCrashed.Result);
+            });
+            Crashes.GetLastSessionCrashReportAsync().ContinueWith(task =>
+            {
+                Log("Crashes.LastSessionCrashReport.Exception=" + task.Result?.Exception);
+            });
+        }
+
+        private static bool ConfirmationHandler()
+        {
+            Current.Dispatcher.Invoke(() =>
```

Comment:
This is blocking until we clicked, it works but it makes logging odd and the intent is to be "asynchronous" here.

Reply:
#1024